synapse/storage/databases/main/pusher.py

```diff
@@ -617,7 +617,7 @@ def set_device_id_for_pushers_txn(txn: LoggingTransaction) -> int:
                 txn=txn,
                 table="pushers",
                 key_names=("id",),
-                key_values=[row[0] for row in rows],
+                key_values=[(row[0],) for row in rows],
```

Comment:
Whoops, is this from the cursor_to_dict changes?

I guess this means we don't have any test coverage for this background update? :(

Reply:
From #16431, yes. Probably not... we should probably fail unit tests if any errors happen in the background, but there's no guarantee this actually runs with any data?